account_analytic_no_lines/models/account_move_line.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2016 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class AccountMoveLine(models.Model):
+
+    _inherit = "account.move.line"
+
+    @api.model
+    def _must_create_analytic_line(self, move_line):
+        if not move_line.product_id:
```

Comment:
Why do you want to do this if you are hidding it? Shouldn't be False by default? Why using expense_policy? In this case, maybe it has sense to add it somewhere else, because expense_policy is defined on `sale`.... WDYT?